fusionauth/resource_fusionauth_user_action.go

```diff
@@ -0,0 +1,336 @@
+			"localized_names": opt.LocalizedNames,
+		})
+	}
+	if err := data.Set("options", options); err != nil {
+		return fmt.Errorf("user_action.options: %s", err.Error())
+	}
+	if err := data.Set("prevent_login", resp.UserAction.PreventLogin); err != nil {
+		return fmt.Errorf("user_action.prevent_login: %s", err.Error())
+	}
+	if err := data.Set("send_end_event", resp.UserAction.SendEndEvent); err != nil {
+		return fmt.Errorf("user_action.send_end_event: %s", err.Error())
+	}
+	if err := data.Set("start_email_template_id", resp.UserAction.StartEmailTemplateId); err != nil {
+		return fmt.Errorf("user_action.start_email_template_id: %s", err.Error())
+	}
+	if err := data.Set("temporal", resp.UserAction.Temporal); err != nil {
+		return fmt.Errorf("user_action.temporal: %s", err.Error())
+	}
+	if err := data.Set("user_emailing_enabled", resp.UserAction.UserEmailingEnabled); err != nil {
+		return fmt.Errorf("user_action.user_emailing_enabled: %s", err.Error())
+	}
+	if err := data.Set("user_notifications_enabled", resp.UserAction.UserNotificationsEnabled); err != nil {
+		return fmt.Errorf("user_action.user_notifications_enabled: %s", err.Error())
+	}
+
+	return nil
+}
+
+func updateUserAction(data *schema.ResourceData, i interface{}) error {
+	client := i.(Client)
+
+	ua := fusionauth.UserAction{}
+
+	if data.HasChange("name") {
+		ua.Name = data.Get("name").(string)
+	}
+	if data.HasChange("cancel_email_template_id") {
+		ua.CancelEmailTemplateId = data.Get("cancel_email_template_id").(string)
+	}
+	if data.HasChange("end_email_template_id") {
+		ua.EndEmailTemplateId = data.Get("end_email_template_id").(string)
+	}
+	if data.HasChange("include_email_in_event_json") {
+		ua.IncludeEmailInEventJSON = data.Get("include_email_in_event_json").(bool)
+	}
+	if data.HasChange("localized_names") {
+		names := data.Get("localized_names").(map[string]interface{})
+		ua.LocalizedNames = make(map[string]string)
+
+		for k, v := range names {
+			ua.LocalizedNames[k] = v.(string)
+		}
+	}
+	if data.HasChange("modify_email_template_id") {
+		ua.ModifyEmailTemplateId = data.Get("modify_email_template_id").(string)
+	}
+	if data.HasChange("options") {
+		ua.Options = buildUserActionOptions(data.Get("options"))
+	}
+	if data.HasChange("prevent_login") {
+		ua.PreventLogin = data.Get("prevent_login").(bool)
+	}
+	if data.HasChange("send_end_event") {
+		ua.SendEndEvent = data.Get("send_end_event").(bool)
+	}
+	if data.HasChange("start_email_template_id") {
+		ua.StartEmailTemplateId = data.Get("start_email_template_id").(string)
+	}
+	if data.HasChange("temporal") {
+		ua.Temporal = data.Get("temporal").(bool)
+	}
+	if data.HasChange("user_emailing_enabled") {
+		ua.UserEmailingEnabled = data.Get("user_emailing_enabled").(bool)
+	}
+	if data.HasChange("user_notifications_enabled") {
+		ua.UserNotificationsEnabled = data.Get("user_notifications_enabled").(bool)
+	}
+
+	resp, faErrs, err := client.FAClient.UpdateUserAction(data.Id(), fusionauth.UserActionRequest{
```

Comment:
Since this is using a `PUT` instead of `PATCH`, I would rebuild the entire `usionauth.UserAction{}`

Reply:
Ah. Is this an upstream issue with their SDK? I'll update my PR to suit.

Reply:
Done!